src/Components/App/App.test.js

```diff
@@ -7,564 +7,697 @@ import { VideoProvider } from '../../Context/VideoContext'
 import { SettingsProvider } from '../../Context/SettingsContext'
 import { SessionProvider } from '../../Context/SessionContext'
 import { MemoryRouter } from 'react-router-dom'
+import { UserProvider, UserContext } from '../../Context/UserContext'
 import { getSettings, updateSettings, getRandomContent, postSession } from '../../apiCalls'
 jest.mock('../../apiCalls.js')
 
 describe('App', () => {
-  it('should bring users to the timer view on load and allow them to visit the Stats, About, and Settings pages', async () => {
-
+  beforeEach(() => {
+    window.gapi = {
+      signin2: {
+        render: jest.fn(),
+      },
+    };
+  });
+
+  it('should render the login page on load', () => {
     getSettings.mockResolvedValue({
       data: {
         id: 1,
-        rest_interval: '5',
-        work_interval: '30',
-        sound: 'chordCliff'
-      }
+        rest_interval: "5",
+        work_interval: "30",
+        sound: "chordCliff",
+      },
     })
-
-    const { getByRole, getByAltText, getByText } = render(
-      <MemoryRouter>
-        <SettingsProvider>
-          <ViewProvider>
+    
+    const { getByRole } = render (
+      <UserProvider>
+        <ViewProvider>
+          <SettingsProvider>
             <SessionProvider>
-              <VideoProvider>
-                <App />
-              </VideoProvider>
+              <App />
             </SessionProvider>
-          </ViewProvider>
-        </SettingsProvider>
-      </MemoryRouter>
+          </SettingsProvider>
+        </ViewProvider>
+      </UserProvider>
     )
 
-    const timerLength = await waitFor(() => getByText(/30:00/i))
-    const appHeading = getByRole('heading', { name: /som timer/i })
-    const timerHeading = getByRole('heading', { name: /click ▶ to begin focusing/i })
+    const welcomePrompt = getByRole('heading', { name: /welcome to som timer/i })
+    const homePagePrompt = getByRole('heading', { name: /home page/i })
+    const aboutPageHeader = getByRole('heading', { name: /about page/i })
+    const settingsPageHeader = getByRole('heading', { name: /settings page/i })
+    const statsPageHeader = getByRole('heading', { name: /stats page/i })
+
+    expect(welcomePrompt).toBeInTheDocument()
+    expect(homePagePrompt).toBeInTheDocument()
+    expect(aboutPageHeader).toBeInTheDocument()
+    expect(settingsPageHeader).toBeInTheDocument()
+    expect(statsPageHeader).toBeInTheDocument()
+  })
+
+  it("should bring users to the timer view on load when a user is logged in and allow them to visit the Stats, About, and Settings pages", async () => {
+    getSettings.mockResolvedValue({
+      data: {
+        id: 1,
+        rest_interval: "5",
+        work_interval: "30",
+        sound: "chordCliff",
+      },
+    });
+
+    const { getByRole, getByAltText, getByText } = render(
+      <MemoryRouter>
+        <UserContext.Provider
```

Comment:
So glad this works! So much easier to mock out the actual gapi function and then just give provider values directly, rather than having to deal with somehow mocking a resolved value for gapi